os/drivers/lcd/lcd_dev.c

```diff
@@ -236,16 +245,21 @@ static int lcddev_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 	}
 	break;
 	case LCDDEVIO_PUTAREA: {
-		FAR const struct lcddev_area_s *lcd_area = (FAR const struct lcddev_area_s *)arg;
-
-		size_t cols = lcd_area->col_end - lcd_area->col_start + 1;
-		size_t pixel_size = priv->planeinfo[lcd_area->planeno].bpp > 1 ? priv->planeinfo[lcd_area->planeno].bpp >> 3 : 1;
-		size_t row_size = lcd_area->stride > 0 ? lcd_area->stride : cols * pixel_size;
+		if (empty == true) {
+			g_lcd_priv = priv;
+			g_lcd_area = (FAR const struct lcddev_area_s *)arg;
+			memcpy(lcd_kbuffer, g_lcd_area->data, CONFIG_LCD_XRES * CONFIG_LCD_YRES * sizeof(uint16_t));
 
-		if (priv->planeinfo[lcd_area->planeno].putarea) {
```

Comment:
how about keep this condition checking here?

Reply:
done